sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -149,34 +146,44 @@ export class SenderImpl implements Sender {
    * @internal
    * @throws Error if the underlying connection is closed.
    */
-  constructor(context: ClientEntityContext, retryOptions: RetryOptions = {}) {
-    throwErrorIfConnectionClosed(context.namespace);
-    this._context = context;
-    this.entityPath = context.entityPath;
-    this._sender = MessageSender.create(this._context, retryOptions);
+  constructor(
+    private _context: ConnectionContext,
+    private _entityPath: string,
+    retryOptions: RetryOptions = {}
+  ) {
+    throwErrorIfConnectionClosed(_context);
+    this.entityPath = _entityPath;
+    this._sender = MessageSender.create(this._context, _entityPath, retryOptions);
+    if (!this._context.managementClients[this._entityPath]) {
```

Comment:
Same comment here as in Receiver - let's move the creation of the management client into a member on this._context (or even simpler - just make it create it on demand when someone asks for a management client for an entity).

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/9710/commits/ed41c32649fd20ca340d73eb8cd032660dc06a15